js/common/lib/training-session.ts

```diff
@@ -0,0 +1,131 @@
+ */
+export interface TrainingSession {
+  // #region run()
+
+  // /**
+  //  * Run TrainStep asynchronously with the given feeds and options.
+  //  *
+  //  * @param feeds - Representation of the model input. See type description of `InferenceSession.InputType` for
+  //  detail.
+  //  * @param options - Optional. A set of options that controls the behavior of model training.
+  //  * @returns A promise that resolves to an error code
+  //  */
+  // runTrainStep(feeds: InferenceSession.FeedsType, options?: InferenceSession.RunOptions):
+  // Promise<InferenceSession.ReturnType>;
+
+  // /**
+  //  * Run a single train step with the given inputs and options.
+  //  *
+  //  * @param feeds - Representation of the model input.
+  //  * @param fetches - Representation of the model output.
+  //  * detail.
+  //  * @param options - Optional. A set of options that controls the behavior of model inference.
+  //  * @returns A promise that resolves to a map, which uses output names as keys and OnnxValue as corresponding
+  //  values.
+  //  */
+  // runTrainStep(feeds: InferenceSession.FeedsType, fetches: InferenceSession.FetchesType,
+  //   options?: InferenceSession.RunOptions): Promise<InferenceSession.ReturnType>;
+
+  // #endregion
+
+  // #region copy parameters
+  // /**
+  //  * Copies from a buffer containing parameters to the TrainingSession parameters.
+  //  * @param buffer - buffer containing parameters
+  //  * @param trainableOnly - Optional. True if trainable parameters only to be modified, false otherwise.
+  //  */
+  // copyParametersFromBuffer(buffer: ArrayBufferLike, trainableOnly: boolean): void;
+
+  // /**
+  //  * Copies from the TrainingSession parameters to the given buffer.
+  //  * @param buffer - buffer that will be modified. Must be the same size as results of GetParametersSize API call.
+  //  * @param trainableOnly - Optional. True if trainable parameters only to be copied, false othrwise.
+  //  */
+  // copyParametersToBuffer(buffer: ArrayBufferLike, trainableOnly: boolean): void;
+  // #endregion
+
+  // #region release()
+
+  /**
+   * Release the inference session and the underlying resources.
+   */
+  release(): Promise<void>;
+  // #endregion
+
+  // #region metadata
+
+  /**
+   * Get input names of the loaded model.
+   */
+  readonly inputNames: readonly string[];
+
+  /**
+   * Get output names of the loaded model.
+   */
+  readonly outputNames: readonly string[];
+  // #endregion
+}
+
+export interface TrainingSessionFactory {
+  // #region create()
+
+  /**
+   * Create a new training session and load models asynchronously from an ONNX model file.
+   *
+   * @param uri - The URI or file path of the model to load.
+   * @param options - specify configuration for creating a new inference session.
+   * @returns A promise that resolves to an InferenceSession object.
+   */
+  create(checkpointStateUri: string, trainModelURI: string, options?: InferenceSession.SessionOptions):
```

Comment:
Dont you need 2 more apis which support ArrayBuffer and Uint8Array option just like below api?

Reply:
added the create calls that process ArrayBuffer & Uint8Array as well as the create calls where checkpoint, trainingmodel, and evalmodel are passed in